### PR TITLE
Added trusts for the heat

### PR DESCRIPTION
**Pull request by cjchang**
Used the trusts for deferred_auth_method and used the heat standard port for haproxy.

### CODE REVIEW

File: roles/heat/tasks/main.yml [hunk 1]
```diff
@@ -9,10 +9,21 @@
 - name: create heat log directory
   file: dest=/var/log/heat state=directory mode=0755 owner=heat
 
+- name: create heat cache dir
+  file: dest=/var/cache/heat state=directory mode=0700
```
Comment (msambol) [inline roles/heat/tasks/main.yml:13]: Can you put each argument on separate lines, indented properly.

File: library/keystone_user.py [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+        # Should never reach here
+        raise ValueError("Code should never reach here")
+
+    return dict(changed=changed, id=id)
+
+# import module snippets
+from ansible.module_utils.basic import *
+from ansible.module_utils.openstack import *
+if __name__ == '__main__':
+    main()
```
Comment (msambol) [inline library/keystone_user.py:431]: Did you write this module yourself or did you get it from somewhere?

File: roles/heat/tasks/main.yml (hunk 1 above)
Comment (cjchang) [inline roles/heat/tasks/main.yml:13]: This code is copied from neutron. Just follow on what is doing today but we can do that.

**Comment by cjchang** [comment]
This is a modified file from 

https://github.com/ansible/ansible-modules-core/blob/devel/cloud/openstack/keystone_user.py

it added a function to allow adding a role only without user. will need to contribute it back to ansible_core.

**Comment by cjchang** [comment]
This is the pull request for keystone_user.py. After this request is pulled, we can remove keystone_user.py from the library.
https://github.com/ansible/ansible-modules-core/pull/1926

**Comment by msambol** [comment]
Sounds good, keep us updated.

File: roles/heat/tasks/main.yml (hunk 1 above)
Comment (cjchang) [inline roles/heat/tasks/main.yml:13]: After reviewing all other components, look like they all have the same lines as they are used here. So I will just keep as is to be consistent with other components for now.

**Pull request merged**